wsaa.py

```diff
@@ -107,112 +111,154 @@ def create_tra(service=SERVICE, ttl=2400):
-
-            try:
-                # Firmar el texto (tra) usando cryptography (openssl bindings para python)
-                p7 = _lib.PKCS7_sign(
-                    cert._x509, private_key._evp_pkey, _ffi.NULL, bio_in, 0
-                )
-            finally:
-                # Liberar memoria asignada
-                _lib.BIO_free(bio_in)
-            # Se crea un buffer nuevo porque la firma lo consume
-            bio_in = _lib.BIO_new_mem_buf(tra, len(tra))
-            try:
-                # Crear buffer de salida
-                bio_out = _lib.BIO_new(_lib.BIO_s_mem())
-                try:
-                    # Instanciar un SMIME
-                    _lib.SMIME_write_PKCS7(bio_out, p7, bio_in, 0)
-
-                    # Tomar datos para la salida
-                    result_buffer = _ffi.new("char**")
-                    buffer_length = _lib.BIO_get_mem_data(bio_out, result_buffer)
-                    p7 = _ffi.buffer(result_buffer[0], buffer_length)[:]
-                finally:
-                    _lib.BIO_free(bio_out)
-            finally:
-                _lib.BIO_free(bio_in)
 
-        else:
-            p7 = pkcs7.PKCS7SignatureBuilder().set_data(
-                tra
-            ).add_signer(
-                cert, private_key, hashes.SHA256()
-            ).sign(
-                serialization.Encoding.SMIME, [pkcs7.PKCS7Options.Binary]
-            )
+def sign_tra_openssl(tra, cert=CERT, privatekey=PRIVATEKEY, passphrase=""):
+    "Workaround using openssl binary directly via command-line interface to sign transactions"
+    try:
+        out = Popen(
+            [
+                openssl_exe(),
+                "smime",
+                "-sign",
+                "-signer",
+                cert,
+                "-inkey",
+                privatekey,
+                "-outform",
+                "DER",
+                "-nodetach",
+            ],
+            stdin=PIPE,
+            stdout=PIPE,
+            stderr=PIPE,
+        ).communicate(tra)[0]
+        return b64encode(out)
+    except OSError as e:
+        if e.errno == 2:
+            warnings.warn("El ejecutable de OpenSSL no esta disponible en el PATH")
+        raise
 
-        # Generar p7 en formato mail y recortar headers
-        msg = email.message_from_string(p7.decode("utf8"))
-        for part in msg.walk():
-            filename = part.get_filename()
-            if filename and filename.startswith("smime.p7"):
-                # Es la parte firmada?
-                # Devolver CMS
-                return part.get_payload(decode=False)
-        else:
-            raise RuntimeError("Part not found")
 
+def sign_tra(tra, cert=CERT, privatekey=PRIVATEKEY, passphrase=""):
+    "Firmar PKCS#7 el TRA y devolver CMS (recortando los headers SMIME)"
+
+    if isinstance(tra, str):
+        tra = tra.encode("utf8")
+
+    if Binding:
+        if sys.version_info.major == 2 or int(cryptography_version[0:2]) < 39:
```

Comment:
Please change this version checking to use split, or it will fail if cryptography isues a release >=100

Reply:
done